modules/database/test/std/rec/addrModifierTest.c

```diff
@@ -0,0 +1,223 @@
+/*************************************************************************\
+* Copyright (c) 2020 Dirk Zimoch
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include <string.h>
+
+#include "dbAccess.h"
+#include "devSup.h"
+#include "alarm.h"
+#include "dbUnitTest.h"
+#include "errlog.h"
+#include "epicsThread.h"
+
+#include "longinRecord.h"
+
+#include "testMain.h"
+
+void recTestIoc_registerRecordDeviceDriver(struct dbBase *);
+
+static void startTestIoc(const char *dbfile)
+{
+    testdbPrepare();
+    testdbReadDatabase("recTestIoc.dbd", NULL, NULL);
+    recTestIoc_registerRecordDeviceDriver(pdbbase);
+    testdbReadDatabase(dbfile, NULL, NULL);
+
+    eltc(0);
+    testIocInitOk();
+    eltc(1);
+}
+
+struct pv {
+    const char *name;
+    long count;
+    double values[4];
+};
+
+static void expectProcSuccess(struct pv *pv)
+{
+    char fieldname[20];
+    testdbPutFieldOk("reset.PROC", DBF_LONG, 1);
+    testDiag("expecting success from %s", pv->name);
+    sprintf(fieldname, "%s.PROC", pv->name);
+    testdbPutFieldOk(fieldname, DBF_LONG, 1);
+    sprintf(fieldname, "%s.SEVR", pv->name);
+    testdbGetFieldEqual(fieldname, DBF_LONG, NO_ALARM);
+    sprintf(fieldname, "%s.STAT", pv->name);
+    testdbGetFieldEqual(fieldname, DBF_LONG, NO_ALARM);
+}
+
+static double initial[] = {0,1,2,3,4,5,6,7,8,9};
+static double buf[10];
```

Comment:
Looks like this variable could become a local of `expectRange()`.

Reply:
Keeping the declaration at the top level, right after that of `initial`, was deliberate because they have to have the same number of elements.

Reply:
Have a look at `NELEMENTS(X)`, which is our name for the `sizeof(X)/sizeof((X)[0])` trick.

```c
#include <dbDefs.h>
...
    double buf[NELEMENTS(initial)];
```

Reply:
Nice. Fixed in 72c5cf6a8e3